ci/svd2rust-regress/src/main.rs

```diff
@@ -13,92 +14,91 @@ use std::path::PathBuf;
 use std::process::{exit, Command};
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::time::Instant;
-use structopt::StructOpt;
 
-#[derive(StructOpt, Debug)]
+#[derive(Parser, Debug)]
 #[structopt(name = "svd2rust-regress")]
 struct Opt {
     /// Run a long test (it's very long)
-    #[structopt(short = "l", long = "long-test")]
+    #[clap(short = 'l', long = "long-test")]
     long_test: bool,
 
     /// Path to an `svd2rust` binary, relative or absolute.
     /// Defaults to `target/release/svd2rust[.exe]` of this repository
     /// (which must be already built)
-    #[structopt(short = "p", long = "svd2rust-path", parse(from_os_str))]
+    #[clap(short = 'p', long = "svd2rust-path")]
     bin_path: Option<PathBuf>,
 
     // TODO: Consider using the same strategy cargo uses for passing args to rustc via `--`
     /// Run svd2rust with `--atomics`
-    #[structopt(long = "atomics")]
+    #[clap(long)]
     atomics: bool,
 
     /// Filter by chip name, case sensitive, may be combined with other filters
-    #[structopt(short = "c", long = "chip", raw(validator = "validate_chips"))]
+    #[clap(short = 'c', long, value_parser = validate_chips)]
     chip: Vec<String>,
 
     /// Filter by manufacturer, case sensitive, may be combined with other filters
-    #[structopt(
-        short = "m",
+    #[clap(
+        short = 'm',
         long = "manufacturer",
-        raw(validator = "validate_manufacturer")
+        value_parser = validate_manufacturer,
     )]
     mfgr: Option<String>,
 
     /// Filter by architecture, case sensitive, may be combined with other filters
     /// Options are: "CortexM", "RiscV", "Msp430", "Mips" and "XtensaLX"
-    #[structopt(
-        short = "a",
+    #[clap(
+        short = 'a',
         long = "architecture",
-        raw(validator = "validate_architecture")
+        value_parser = validate_architecture,
     )]
     arch: Option<String>,
 
     /// Include tests expected to fail (will cause a non-zero return code)
-    #[structopt(short = "b", long = "bad-tests")]
+    #[clap(short = 'b', long = "bad-tests")]
     bad_tests: bool,
 
     /// Enable formatting with `rustfmt`
-    #[structopt(short = "f", long = "format")]
+    #[clap(short = 'f', long)]
     format: bool,
 
     /// Print all available test using the specified filters
-    #[structopt(long = "list")]
+    #[clap(long)]
     list: bool,
 
     /// Path to an `rustfmt` binary, relative or absolute.
     /// Defaults to `$(rustup which rustfmt)`
-    #[structopt(long = "rustfmt_bin_path", parse(from_os_str))]
+    #[clap(long = "rustfmt_bin_path")]
```

Comment:
```suggestion
    #[clap(long)]
```

`svd2rust` used, with `structopt`, snake case

Reply:
?

Reply:
the flag should be `--rustfmt-bin-path` not `--rustfmt_bin_path`

Reply:
done, rebased